Userland/Applications/PixelPaint/Tools/LassoSelectTool.cpp

```diff
@@ -138,15 +137,21 @@ void LassoSelectTool::flood_lasso_selection(Gfx::Bitmap& lasso_bitmap, Gfx::IntP
 
 void LassoSelectTool::on_second_paint(Layer const* layer, GUI::PaintEvent& event)
 {
-    if (!m_selecting)
+    if (!m_selecting || m_preview_coords.size() < 2)
         return;
     GUI::Painter painter(*m_editor);
     painter.add_clip_rect(event.rect());
     if (layer)
         painter.translate(editor_layer_location(*layer));
+
+    Gfx::Path preview_path;
+    preview_path.move_to(editor_stroke_position(m_preview_coords.first(), 1).to_type<float>());
+    for (auto it = ++m_preview_coords.begin(); it != m_preview_coords.end(); ++it) {
+        preview_path.line_to(editor_stroke_position(*it, 1).to_type<float>());
+    }
```

Comment:
Since you're now making the `Gfx::Path` in the `on_second_paint()` rather than incrementally, I think this should be painted like the polygon selection tool (two calls to `draw_lines()`). Using a `Gfx::Path` is pretty much allocating a vector again, which I think would be nicer avoided :)

Reply:
Done, no idea why I didn't do it this way, since I've only just done the polygonal select tool. Thanks :)